src/toil/batchSystems/mesos/executor.py

```diff
@@ -102,12 +147,33 @@ def launchTask(self, driver, task):
         def runTask():
             log.debug("Running task %s", task.task_id.value)
             sendUpdate(mesos_pb2.TASK_RUNNING)
-            # This is where task.data is first invoked. Using this position to setup cleanupInfo
             taskData = pickle.loads(task.data)
             if self.workerCleanupInfo is not None:
                 assert self.workerCleanupInfo == taskData.workerCleanupInfo
+                assert self.masterAddress == taskData.masterAddress
+                assert self.mesosCredentials == taskData.mesosCredentials
             else:
+                # This is where task.data is invoked for the firs ttiem on this executor. Using this
+                # position to setup workerCleanupInfo, masterAddress, mesosCredentials, and
+                # deadSpace for this executor.
                 self.workerCleanupInfo = taskData.workerCleanupInfo
+                self.masterAddress = taskData.masterAddress
```

Comment:
I couldn't find documentation anywhere. I'd assume the driver knows this. I know that the `MesosExecutorDriver` object has no `__dict__` so I couldn't find the required info by looking there.